crates/syntax/src/lib.rs

```diff
@@ -212,6 +212,13 @@ impl ast::Attr {
     }
 }
 
+impl ast::Stmt {
+    /// Returns `text`, parsed as statement, but only if it has no errors.
+    pub fn parse(text: &str) -> Result<Self, ()> {
```

Comment:
Fair point. Doesn't seem like every type (`ast::Attr` right above) has an explicit test though.

Not sure what to add for the `err` case. Could add some garbage, but I guess we rather want bits of code that should explicitly _not_ be parseable as `Stmt`. Suggestions?

Reply:
Ah, I hadn't seen the parser for `ast::Attr`, that must have been added since I implemented the others. I was thinking of the `ast::Item::parse` etc, which have tests - e.g. `item_parser_tests`.

For error cases, I guess:
- Something that isn't a statement. e.g. and attribute `#[foo]`
- Multiple statements: `a(); b();`
- Something unparsable: `)`

Reply:
Just letting you know in case you feel like adding more tests to the other parsers.

Apart your suggestions I've added a bunch of items, expressions and blocks to the mix that can be parsed as `Stmt`.